Applications/Games/Space Engineers/Steam/script.js

```diff
@@ -0,0 +1,32 @@
+include(["engines", "wine", "quick_script", "steam_script"]);
+include(["engines", "wine", "verbs", "dotnet461"]);
+include(["engines", "wine", "verbs", "vcrun2015"]);
+include(["engines", "wine", "verbs", "dxvk"]);
+include(["engines", "wine", "verbs", "xact"]);
+
+var installerImplementation = {
+    run: function () {
+        new SteamScript()
+            .name("Space Engineers")
+            .editor("Keen Software House")
+            .author("Zemogiter")
+            .appId(244850)
+            .wineVersion("3.20")
+            .wineDistribution("staging")
+            .wineArchitecture("amd64")
+            .preInstall(function (wine, wizard) {
+                if (var operatingSystem != LINUX) {
+                    break;
```

Comment:
After looking at how `var operatingSystemFetcher` in `Installer Script` is declared I was sure it would work but:
```
javax.script.ScriptException: <eval>:22:20 Expected an operand but found var
                if (var operatingSystem != LINUX) {
                    ^ in <eval> at line number 22 at column number 20
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.throwAsScriptException(NashornScriptEngine.java:477)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.compileImpl(NashornScriptEngine.java:544)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.compileImpl(NashornScriptEngine.java:531)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.evalImpl(NashornScriptEngine.java:409)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.eval(NashornScriptEngine.java:162)
	at java.scripting/javax.script.AbstractScriptEngine.eval(AbstractScriptEngine.java:264)
	at org.phoenicis.scripts/org.phoenicis.scripts.nashorn.NashornEngine.eval(NashornEngine.java:50)
	at org.phoenicis.scripts/org.phoenicis.scripts.nashorn.NashornScriptInterpreter.runScript(NashornScriptInterpreter.java:35)
	at org.phoenicis.scripts/org.phoenicis.scripts.interpreter.BackgroundScriptInterpreter.lambda$runScript$0(BackgroundScriptInterpreter.java:35)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: jdk.nashorn.internal.runtime.ParserException: <eval>:22:20 Expected an operand but found var
                if (var operatingSystem != LINUX) {
                    ^
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.AbstractParser.error(AbstractParser.java:297)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.AbstractParser.error(AbstractParser.java:282)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.unaryExpression(Parser.java:4462)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.expression(Parser.java:4620)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.conditionalExpression(Parser.java:4772)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.assignmentExpression(Parser.java:4711)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.expression(Parser.java:4589)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.expression(Parser.java:4585)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.ifStatement(Parser.java:1883)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.statement(Parser.java:1048)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.sourceElements(Parser.java:909)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.functionBody(Parser.java:4206)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.functionExpression(Parser.java:3831)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.memberExpression(Parser.java:3578)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.leftHandSideExpression(Parser.java:3407)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.unaryExpression(Parser.java:4440)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.expression(Parser.java:4620)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.conditionalExpression(Parser.java:4772)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.assignmentExpression(Parser.java:4711)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.argumentList(Parser.java:3728)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.leftHandSideExpression(Parser.java:3430)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.unaryExpression(Parser.java:4440)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.expression(Parser.java:4620)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.conditionalExpression(Parser.java:4772)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.assignmentExpression(Parser.java:4711)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.expression(Parser.java:4589)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.expression(Parser.java:4585)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.expressionStatement(Parser.java:1854)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.statement(Parser.java:1155)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.sourceElements(Parser.java:909)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.functionBody(Parser.java:4206)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.functionExpression(Parser.java:3831)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.memberExpression(Parser.java:3578)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.leftHandSideExpression(Parser.java:3407)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.unaryExpression(Parser.java:4440)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.expression(Parser.java:4620)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.conditionalExpression(Parser.java:4772)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.assignmentExpression(Parser.java:4711)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.propertyAssignment(Parser.java:3227)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.objectLiteral(Parser.java:3001)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.primaryExpression(Parser.java:2801)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.memberExpression(Parser.java:3623)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.leftHandSideExpression(Parser.java:3407)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.unaryExpression(Parser.java:4440)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.expression(Parser.java:4620)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.conditionalExpression(Parser.java:4772)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.assignmentExpression(Parser.java:4711)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.variableDeclarationList(Parser.java:1633)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.variableStatement(Parser.java:1532)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.statement(Parser.java:1042)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.sourceElements(Parser.java:909)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.program(Parser.java:844)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.parse(Parser.java:325)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.parse(Parser.java:285)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.Context.compile(Context.java:1500)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.Context.compileScript(Context.java:1467)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.Context.compileScript(Context.java:750)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.compileImpl(NashornScriptEngine.java:542)
	... 10 more
```

So I've replaced var with this.operatingSystem but it gave me a different error:
```
javax.script.ScriptException: <eval>:24:20 Illegal break statement
                    break;
                    ^ in <eval> at line number 24 at column number 20
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.throwAsScriptException(NashornScriptEngine.java:477)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.compileImpl(NashornScriptEngine.java:544)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.compileImpl(NashornScriptEngine.java:531)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.evalImpl(NashornScriptEngine.java:409)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.eval(NashornScriptEngine.java:162)
	at java.scripting/javax.script.AbstractScriptEngine.eval(AbstractScriptEngine.java:264)
	at org.phoenicis.scripts/org.phoenicis.scripts.nashorn.NashornEngine.eval(NashornEngine.java:50)
	at org.phoenicis.scripts/org.phoenicis.scripts.nashorn.NashornScriptInterpreter.runScript(NashornScriptInterpreter.java:35)
	at org.phoenicis.scripts/org.phoenicis.scripts.interpreter.BackgroundScriptInterpreter.lambda$runScript$0(BackgroundScriptInterpreter.java:35)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: jdk.nashorn.internal.runtime.ParserException: <eval>:24:20 Illegal break statement
                    break;
                    ^
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.AbstractParser.error(AbstractParser.java:272)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.AbstractParser.error(AbstractParser.java:256)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.breakStatement(Parser.java:2298)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.statement(Parser.java:1063)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.statement(Parser.java:1028)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.statementList(Parser.java:1463)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.getBlock(Parser.java:612)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.getStatement(Parser.java:637)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.getStatement(Parser.java:632)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.ifStatement(Parser.java:1885)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.statement(Parser.java:1048)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.sourceElements(Parser.java:909)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.functionBody(Parser.java:4206)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.functionExpression(Parser.java:3831)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.memberExpression(Parser.java:3578)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.leftHandSideExpression(Parser.java:3407)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.unaryExpression(Parser.java:4440)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.expression(Parser.java:4620)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.conditionalExpression(Parser.java:4772)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.assignmentExpression(Parser.java:4711)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.argumentList(Parser.java:3728)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.leftHandSideExpression(Parser.java:3430)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.unaryExpression(Parser.java:4440)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.expression(Parser.java:4620)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.conditionalExpression(Parser.java:4772)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.assignmentExpression(Parser.java:4711)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.expression(Parser.java:4589)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.expression(Parser.java:4585)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.expressionStatement(Parser.java:1854)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.statement(Parser.java:1155)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.sourceElements(Parser.java:909)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.functionBody(Parser.java:4206)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.functionExpression(Parser.java:3831)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.memberExpression(Parser.java:3578)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.leftHandSideExpression(Parser.java:3407)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.unaryExpression(Parser.java:4440)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.expression(Parser.java:4620)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.conditionalExpression(Parser.java:4772)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.assignmentExpression(Parser.java:4711)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.propertyAssignment(Parser.java:3227)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.objectLiteral(Parser.java:3001)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.primaryExpression(Parser.java:2801)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.memberExpression(Parser.java:3623)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.leftHandSideExpression(Parser.java:3407)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.unaryExpression(Parser.java:4440)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.expression(Parser.java:4620)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.conditionalExpression(Parser.java:4772)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.assignmentExpression(Parser.java:4711)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.variableDeclarationList(Parser.java:1633)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.variableStatement(Parser.java:1532)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.statement(Parser.java:1042)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.sourceElements(Parser.java:909)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.program(Parser.java:844)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.parse(Parser.java:325)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.parse(Parser.java:285)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.Context.compile(Context.java:1500)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.Context.compileScript(Context.java:1467)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.Context.compileScript(Context.java:750)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.compileImpl(NashornScriptEngine.java:542)
	... 10 more
```

Reply:
Should be something like:
```Javascript
var operatingSystemFetcher = Bean("operatingSystemFetcher");
if (operatingSystemFetcher.fetchCurrentOperationSystem() !== "Linux")
```

Reply:
Applied the change but it complains about illegal break. I've checked on w3schools and this break should work.

Reply:
What do you want to do? Preventing preinstall on other system (aka OSX) that Linux? If so, try return instead of break

Reply:
@qparis I want to prevent the whole script from runing on OSX.

Reply:
@Zemogiter Does the game run without DXVK ? If yes, you could simply install DXVK only if the OS is linux, and still allow the script to be run on macOS.

Reply:
@ImperatorS79 no this game needs DXVK (because of transform feedback) and since MoltenVK is yet to be merged into Wine, we can't let users run this script on OSX or we get people complaining about stuff we cannot fix

Reply:
@Zemogiter Why not simply using compatible operating system flag on the json file?

Reply:
@qparis because even when checking the "All operating systems" it's still being displayed

Reply:
Well, in this case, it probably means that the fact that « All Operating Systems » scripts are not compatible and that they are accessible for testing purpose should be made clearer.
I suggest we open an issue for this